src/delegate/client.rs

```diff
@@ -23,17 +23,21 @@ use super::not_initialized_error;
 /// Maps all pending client request IDs to their future responses.
 type RequestMap = DashMap<u64, oneshot::Sender<Output>>;
 
-/// Handle for communicating with the language client.
+/// Internals for language client handle.
 #[derive(Debug)]
-pub struct Client {
+pub(super) struct ClientInner {
```

Comment:
I don't think this struct needs to be `pub(super)` since it's never supposed accessed from the outside. Mind if you could change this to be private?